jekyll/_cci2/nomad-metrics.md

```diff
@@ -0,0 +1,225 @@
+sudo service docker restart
+sleep 5
+
+echo "--------------------------------------"
+echo "         Installing nomad"
+echo "--------------------------------------"
+apt-get install -y zip
+curl -o nomad.zip https://releases.hashicorp.com/nomad/0.5.6/nomad_0.5.6_linux_amd64.zip
+unzip nomad.zip
+mv nomad /usr/bin
+
+echo "--------------------------------------"
+echo "      Creating config.hcl"
+echo "--------------------------------------"
+export PRIVATE_IP="$(/sbin/ifconfig eth0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}')"
+mkdir -p /etc/nomad
+cat <<EOT > /etc/nomad/config.hcl
+log_level = "DEBUG"
+
+data_dir = "/opt/nomad"
+datacenter = "us-east-1"
+
+advertise {
+    http = "$PRIVATE_IP"
+    rpc = "$PRIVATE_IP"
+    serf = "$PRIVATE_IP"
+}
+
+client {
+    enabled = true
+
+    # Expecting to have DNS record for nomad server(s)
+    servers = ["$SERVICES_PRIVATE_IP:4647"]
+    node_class = "linux-64bit"
+    options = {"driver.raw_exec.enable" = "1"}
+}
+
+telemetry {
+    publish_node_metrics = true
+    statsd_address = "$SERVICES_PRIVATE_IP:8125"
+}
+EOT
+
+echo "--------------------------------------"
+echo "      Creating nomad.conf"
+echo "--------------------------------------"
+cat <<EOT > /etc/init/nomad.conf
+start on filesystem or runlevel [2345]
+stop on shutdown
+
+script
+    exec nomad agent -config /etc/nomad/config.hcl
+end script
+EOT
+
+echo "--------------------------------------"
+echo "   Creating ci-privileged network"
+echo "--------------------------------------"
+docker network create --driver=bridge --opt com.docker.network.bridge.name=ci-privileged ci-privileged
+
+echo "--------------------------------------"
+echo "      Starting Nomad service"
+echo "--------------------------------------"
+service nomad restart
+
+echo "--------------------------------------"
+echo "      Setting up Nomad metrics"
+echo "--------------------------------------"
+docker pull $CONTAINER_IMAGE
+docker rm -f $CONTAINER_NAME || true
+
+# Not using --detach so that upstart can perform log management and process
+# monitoring
+docker run -d --name $CONTAINER_NAME \
+    --rm \
+    --net=host \
+    --userns=host \
+    $CONTAINER_IMAGE \
+    start --nomad-uri=http://localhost:4646 --statsd-host=$SERVICES_PRIVATE_IP --statsd-port=$NOMAD_METRICS_PORT --client
+
+```
+
+#### Updating the Auto Scaling Group
+
+1. Select the `Auto Scaling Groups` (ASG) link in the the sidebar on the left.
+2. Select the ASG with a name tag similar to`*_nomad_clients_asg`
+3. In the description box at the bottom, select the `Edit` button
+4. Select the newly created Launch Configuration from the drop-down.
+5. Press the `Save` button.
+6. You will need to terminate the existing Nomad Client instances in order for newer instances with the new Launch Configuration to be created.  Ideally, this should only be done during off-hours or inside of a maintenance window as it could prematurely terminate running jobs.
```

Comment:
Point 6 should be a new section `Final Steps` that calls out what needs to happen next. It might also we good to place a warning at the top that indicates this will involve downtime in case someone just starts going through the list without reading to the very end.